pkgs/by-name/cr/cratedb/package.nix

```
@@ -0,0 +1,80 @@
+{
+  lib,
+  stdenv,
+  maven,
+  fetchFromGitHub,
+  replaceVars,
+  openjdk23,
+  libarchive,
+  makeWrapper,
+}:
+let
+  # Wants at least Java 22
+  jdk = openjdk23;
+  version = "5.9.3";
+in
+maven.buildMavenPackage {
+  pname = "cratedb";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "crate";
+    repo = "crate";
+    tag = version;
+    hash = "sha256-k+b11pU6ct1s6FEKW3j8kNl+3hYTb+SutcjBI6VKI5s=";
+  };
+
+  nativeBuildInputs = [
+    libarchive
+    makeWrapper
+  ];
+
+  patches = [
+    (replaceVars ./fix-poms.patch { inherit jdk; })
+  ];
+
+  mvnHash = "sha256-u24+hYGY+zE93IhMTqj+Bdzuks6yFFA1uSk/Ya5mxLc=";
+  mvnJdk = jdk;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
+
+    # Don't install the bundled JDK; symlink instead.
+    # Fixing all the paths in the script is frankly way too much work.
+    bsdtar -xf app/target/crate-${version}.tar.gz -C $out --exclude="*/jdk/*" --strip-components=1
+    ln -s ${jdk} $out/jdk
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    # By default it wants to write to $out/data and $out/logs. Bad program.
+    wrapProgram $out/bin/crate \
+      --set CRATE_DISABLE_GC_LOGGING 1 \
+      --add-flags '-Cpath.data="''${XDG_DATA_HOME:-$HOME/.local/share}/crate"' \
+      --add-flags '-Cpath.logs="''${XDG_STATE_HOME:-$HOME/.local/state}/crate"'
+  '';
+
+  meta = {
+    description = "Distributed and scalable SQL database";
+    longDescription = ''
+      CrateDB is a distributed SQL database that makes it simple to store and analyze massive
+      amounts of data in real-time.
+
+      CrateDB offers the benefits of an SQL database and the scalability and flexibility typically
+      associated with NoSQL databases. Modest CrateDB clusters can ingest tens of thousands of records
+      per second without breaking a sweat. You can run ad-hoc queries using standard SQL.
+      CrateDB's blazing-fast distributed query execution engine parallelizes query workloads across
+      the whole cluster.
+    '';
+    homepage = "https://cratedb.com/database";
+    changelog = "https://cratedb.com/docs/crate/reference/en/latest/appendices/release-notes/${version}.html";
+    license = with lib.licenses; [ asl20 ];
+    maintainers = with lib.maintainers; [ pluiedev ];
+    mainProgram = "crate";
+    # Deps are broken on Darwin
+    broken = stdenv.hostPlatform.isDarwin;
```

```suggestion
    broken = stdenv.hostPlatform.system != "x86_64-linux";
```
Don't know how else to express this well. Seems the maven deps are architecture dependent :/

---

I think I'm just going to set the platform to `x86_64-linux` only for now